### PR TITLE
Add support of CancellationToken

### PR DESCRIPTION
**Pull request by ylabade**
**What issue does this PR address?**

- Support of CancellationToken #4557

**Does this PR introduce a breaking change?**
Yes

**Please check if the PR fulfills these requirements**
- [ ] The commit follows our [guidelines](https://github.com/IdentityServer/IdentityServer4/blob/main/.github/CONTRIBUTING.md)
- [ ] Unit Tests for the changes have been added (for bug fixes / features)

**Other information**:

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/IdentityServer/IdentityServer4?pullRequest=4558) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/IdentityServer/IdentityServer4?pullRequest=4558) before we can accept your contribution.<br/><br/>:x: ylabade [sign now](https://cla.dotnetfoundation.org/IdentityServer/IdentityServer4?pullRequest=4558)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/IdentityServer/IdentityServer4?pullRequest=4558) it.</sub>

File: src/IdentityServer4/src/Hosting/IdentityServerMiddleware.cs [hunk 1]
```diff
@@ -51,7 +52,9 @@ public async Task Invoke(HttpContext context, IEndpointRouter router, IUserSessi
                 {
                     _logger.LogInformation("Invoking IdentityServer endpoint: {endpointType} for {url}", endpoint.GetType().FullName, context.Request.Path.ToString());
 
-                    var result = await endpoint.ProcessAsync(context);
+                    var cancellationToken = context.Features.Get<IHttpRequestLifetimeFeature>()?.RequestAborted ?? default;
```
Comment (Sciumo) [inline src/IdentityServer4/src/Hosting/IdentityServerMiddleware.cs:55]: The ability for **RequestAborted** appears to be the main capability added. Isn't there a way to provide access through just the middleware without passing in cancellationtoken everywhere?  Also, I don't see any tests for possible insecure states cancellation may cause.

Comment (ylabade) [reply src/IdentityServer4/src/Hosting/IdentityServerMiddleware.cs:55]: Thank for your feedback. It's not possible to use a middleware in this case and it's important to propagate the cancellation token to the lowest layers to cancel all operations (sql queries, http calls etc.)

**Comment by ylabade** [comment]
@brockallen I don't understand why all projects reference IdentityServer4 dependencies as package reference and not as project reference.

https://github.com/IdentityServer/IdentityServer4/blob/03e56b4cbadb89608833f3dbdc3837c7486f8869/src/IdentityServer4/src/IdentityServer4.csproj#L43

That's why builds fail.

Thank you for your support.

**Comment by leastprivilege** [comment]
We don't have immediate plans to do this as this is a big breaking change. I leave this open for future consideration.

Thanks!

**Comment by ylabade** [comment]
Is there any chance to re-open this PR and add it to the milestone 5.0 ?

Thanks!

**Pull request closed without merging**